src/OpenApi/gen/XmlCommentGenerator.Emitter.cs

```diff
@@ -0,0 +1,392 @@
+    using System.Reflection;
+    using System.Text.Json;
+    using System.Text.Json.Nodes;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.OpenApi;
+    using Microsoft.AspNetCore.Mvc.Controllers;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.OpenApi.Models;
+    using Microsoft.OpenApi.Any;
+
+    {{GeneratedCodeAttribute}}
+    file record XmlComment(
+        string? Summary,
+        string? Description,
+        string? Remarks,
+        string? Returns,
+        string? Value,
+        bool Deprecated,
+        List<string>? Examples,
+        List<XmlParameterComment>? Parameters,
+        List<XmlResponseComment>? Responses);
+
+    {{GeneratedCodeAttribute}}
+    file record XmlParameterComment(string? Name, string? Description, string? Example, bool Deprecated);
+
+    {{GeneratedCodeAttribute}}
+    file record XmlResponseComment(string Code, string? Description, string? Example);
+
+    {{GeneratedCodeAttribute}}
+    file static class XmlCommentCache
+    {
+        private static Dictionary<(Type?, string?), XmlComment>? _cache;
+        public static Dictionary<(Type?, string?), XmlComment> Cache
+        {
+            get
+            {
+                if (_cache is null)
+                {
+                    _cache = GenerateCacheEntries();
+                }
+                return _cache;
+            }
+        }
+
+        private static Dictionary<(Type?, string?), XmlComment> GenerateCacheEntries()
+        {
+            var _cache = new Dictionary<(Type?, string?), XmlComment>();
+{{commentsFromXmlFile}}
+{{commentsFromCompilation}}
+            return _cache;
+        }
+    }
+
+    {{GeneratedCodeAttribute}}
+    file class XmlCommentOperationTransformer : IOpenApiOperationTransformer
+    {
+        public Task TransformAsync(OpenApiOperation operation, OpenApiOperationTransformerContext context, CancellationToken cancellationToken)
+        {
+            var methodInfo = context.Description.ActionDescriptor is ControllerActionDescriptor controllerActionDescriptor
+                ? controllerActionDescriptor.MethodInfo
+                : context.Description.ActionDescriptor.EndpointMetadata.OfType<MethodInfo>().SingleOrDefault();
+
+            if (methodInfo is null)
+            {
+                return Task.CompletedTask;
+            }
+            if (XmlCommentCache.Cache.TryGetValue((methodInfo.DeclaringType, methodInfo.Name), out var methodComment))
+            {
+                if (methodComment.Summary is not null)
+                {
+                    operation.Summary = methodComment.Summary;
+                }
+                if (methodComment.Description is not null)
+                {
+                    operation.Description = methodComment.Description;
+                }
+                if (methodComment.Remarks is not null)
+                {
+                    operation.Description = methodComment.Remarks;
```

Comment:
```suggestion
                    operation.Remarks = methodComment.Remarks;
```

Reply:
There's no `Remarks` property on an OpenAPI operation. The decision to apply `<remarks>` as Description is a bit opionated here but we have flexibility to change it.

Reply:
Ahhh. I thought it was a copy-pasta snafu.